internal/controller/replication.storage/status.go

```diff
@@ -65,6 +65,40 @@ func setFailedPromotionCondition(conditions *[]metav1.Condition, observedGenerat
 		ObservedGeneration: observedGeneration,
 		Status:             metav1.ConditionFalse,
 	})
+	setStatusCondition(conditions, &metav1.Condition{
```

Comment:
This doesnt look correct, even during the promotion failure there could be a state where the Precondition is also not met.

Reply:
> This doesnt look correct, even during the promotion failure there could be a state where the Precondition is also not met.

Let's fix it when we are verifying all the conditions in the future.
As you already mentioned in your new tracker, not all conditions currently expressed maybe very granular and correct right now.